<launch>
	<arg name="veh"/>
	<arg name="local" default="false" doc="true to launch locally on laptop. false to launch of vehicle"/>
	<arg name="pkg_name" default="state_pose_publisher" doc="name of the package"/>
	<arg name="node_name" default="state_pose_publisher_node" doc="name of the node"/>
	<arg name="tango" default="false" doc="use tango offline pose"/>
	<arg name="tango_ros" default="false" doc="use tango real timepose"/>	
	<arg name="scale" default="1" doc="pose scale"/>
	<arg name="shift_x" default="0" doc="pose initial x"/>
	<arg name="shift_y" default="0" doc="pose initial y"/>

	<group ns="$(arg veh)">
	<!-- Local -->

		<remap if="$(arg tango_ros)" from="state_pose_publisher_node/pose" to="/tango_sensors/tango_pose"/>
		<remap unless="$(arg tango_ros)" from="state_pose_publisher_node/pose" to="L2R_pose_pose_node/pose"/>

		<node if="$(arg local)" pkg="$(arg pkg_name)" type="$(arg node_name).py" name="$(arg node_name)" output="screen" clear_params="true" required="true">
		<param name="tango" type="string" value="$(arg tango)"/>
		<param name="scale" type="double" value="$(arg scale)"/>
		<param name="shift_x" type="double" value="$(arg shift_x)"/>
		<param name="shift_y" type="double" value="$(arg shift_y)"/>
		</node>
	<!-- Remote -->
		<include unless="$(arg local)" file="$(find duckietown)/machines"/>
		<node unless="$(arg local)" machine="$(arg veh)" pkg="$(arg pkg_name)" type="$(arg node_name).py" name="$(arg node_name)" output="screen" clear_params="true" required="true">
		<param name="tango" type="string" value="$(arg tango)"/>
		<param name="scale" type="double" value="$(arg scale)"/>
		<param name="shift_x" type="double" value="$(arg shift_x)"/>
		<param name="shift_y" type="double" value="$(arg shift_y)"/>		
		</node>
	</group>
</launch>
